ixx/src/action/index.rs

```diff
@@ -53,52 +53,82 @@ pub(crate) fn index(module: IndexModule) -> anyhow::Result<()> {
   let mut index = Index::default();
   raw_options.keys().for_each(|name| index.push(name));
 
-  println!("Writing index to {}", module.output.to_string_lossy());
+  println!("Writing index to {}", module.index_output.to_string_lossy());
 
-  let mut output = File::create(module.output)?;
+  let mut output = File::create(module.index_output)?;
   index.write_into(&mut output)?;
 
-  println!("Writing meta");
+  println!("Writing meta to {}", module.meta_output.to_string_lossy());
 
-  if !Path::new("meta").exists() {
-    std::fs::create_dir("meta")?;
+  if !module.meta_output.exists() {
+    std::fs::create_dir(&module.meta_output)?;
   }
 
   let options: Vec<libixx::Option> = raw_options.into_values().collect();
   for (idx, chunk) in options.chunks(module.chunk_size).enumerate() {
-    let mut file = File::create(format!("meta/{}.json", idx))?;
+    let mut file = File::create(module.meta_output.join(format!("{}.json", idx)))?;
     serde_json::to_writer(&mut file, &chunk)?;
   }
 
   Ok(())
 }
 
-fn into_option(name: &str, option: option::Option) -> libixx::Option {
-  libixx::Option {
+fn into_option(
+  url_prefix: &Url,
+  name: &str,
+  option: option::Option,
+) -> Result<libixx::Option, url::ParseError> {
+  Ok(libixx::Option {
     declarations: option
       .declarations
       .into_iter()
-      .map(update_declaration)
-      .collect(),
+      .map(|declaration| update_declaration(url_prefix, declaration))
+      .collect::<Result<_, _>>()?,
     default: option.default.map(|option| option.render()),
     description: option.description,
     example: option.example.map(|example| example.render()),
     read_only: option.read_only,
     r#type: option.r#type,
     name: name.to_string(),
+  })
+}
+
+fn update_declaration(url_prefix: &Url, declaration: Declaration) -> Result<Url, url::ParseError> {
+  match declaration {
+    Declaration::StorePath(path) => url_prefix.join(path.split_at(51).1),
+    Declaration::Url { name: _, url } => Ok(url),
   }
 }
 
-fn update_declaration(declaration: Declaration) -> String {
-  // NOTE: Is the url actually optional? If its true, this can be ignored.
-  //       Otherwise the fallback with building the url outself is required.
-  //
-  // if "url" in declaration:
-  //   return declaration["url"]
-  // if declaration.startswith("/nix/store/"):
-  //   # strip prefix: /nix/store/0a0mxyfmad6kaknkkr0ysraifws856i7-source
-  //   return f"{url}{declaration[51:]}"
-  // return declaration
-
-  declaration.url
+#[cfg(test)]
+mod test {
+  use url::Url;
+
+  use crate::{action::index::update_declaration, option::Declaration};
+
+  #[test]
+  fn test_update_declaration() {
+    assert_eq!(
+      update_declaration(
+        &Url::parse("https://nuschtos.de").unwrap(),
+        Declaration::StorePath(
+          "/nix/store/vgvk6q3zsjgb66f8s5cm8djz6nmcag1i-source/modules/initrd.nix".to_string()
```

Comment:
```suggestion
          "/nix/store/vgvk6q3zsjgb66f8s5cm8djz6nmcag1i-source-two/modules/initrd.nix".to_string()
```

that doesn't work with the 51 chars hardcoded from above?

Reply:
Wasn't it always just "source"?

Reply:
No, that's the name of the derivation which is by default source but could be anything except / IIRC